### ISSUE TITLE
`excludePositionTarget` property as a public API

### ISSUE THREAD

**Issue by tkiryu** [IgniteUI/igniteui-angular#4605]
## Is your feature request related to a problem? Please describe.
#4470. I have the same issue with my drop down component using `igx-drop-down`.

Here is a sample.
https://stackblitz.com/edit/igx-drop-down-opening-issue

### Steps to reproduce
1. Run the sample above.
2. Click arrow down button.
3. Click arrow up button.

### Resut
After step 3, drop-down opened again.

### Expected result
After step 3, drop-down should close.

## Describe the solution you'd like
I'd like to use `excludePositionTarget` property to solve the problem.

Thanks to the [fix for #4470](https://github.com/IgniteUI/igniteui-angular/commit/1c197599dc234cea316f13322335073ee3962d50), I found out that I can solve the problem by setting `excludePositionTarget: true` .

But, currently,  `excludePositionTarget` property is not exposed as a public API.
https://github.com/IgniteUI/igniteui-angular/blob/7.2.6/projects/igniteui-angular/src/lib/services/overlay/utilities.ts#L55-L59

Could you expose `excludePositionTarget`? Or, any concerns about doing that?

**Comment by wnvko** [comment]
`excludePositionTarget` property was added for internal use only. We are using it in [`igxToggleAction`](https://github.com/IgniteUI/igniteui-angular/blob/63bf660a33bee7f833dbd7d377c2b19f9426e564/projects/igniteui-angular/src/lib/directives/toggle/toggle.directive.ts#L418). This allows user to close the toggle by clicking on `igxToggleAction`'s target, when `closeOnOutsideClick` is set to `true`. If anyone need to exclude the target he can use `igxToggleAction` - this will automatically set `excludePositionTarget` to `true`. This is why we prefer to retain this property as private.
Please note, `igxOverlay` has no option allowing you to exclude other elements from `closeOnOutsideClick`. For example imagine that you have a button somewhere in the page and an input on totally different place in the same page. On button click you are opening a toggle (or drop down) and the target of the toggle is the input. Right now there is no way to exclude the button from close on outside click as it is not the target. If this is what you need please log a new issue - feature request and we will look into it.

**Comment by tkiryu** [comment]
@wnvko , thank you for explaining why `excludePositionTarget ` is private and we can use `igxToggleAction` instead. I totally understand!

I'm closing it.
